### ISSUE TITLE
Custom Wrapper fires the onMouseEnter twice

### ISSUE THREAD

**Issue by vforge** [styleguidist/react-styleguidist#1255]
**Current behavior**

A very simple custom `Wrapper` causes component's `onMouseOver` to fire twice.

**To reproduce**

Check console when you hover over Button example on https://vforge.github.io/react-styleguidist-mouse-enter-bug/

Source code: https://github.com/vforge/react-styleguidist-mouse-enter-bug

**Expected behavior**

`onMouseEnter` should fire only once.

**Comment by vforge** [comment]
Is there anything else I can provide in order to reproduce this bug?

**Comment by devonmitton** [comment]
I can confirm I'm seeing the same thing. I have a tooltip component that was firing mouseEnter twice when it should only happen once. Only happens when the component is in Styleguidist. Using the component in an app works as expected.

I'm on Styleguidist 7.3.11. I'm in the process of upgrading to the latest version, and I'll follow-up once I've done that.

**Comment by devonmitton** [comment]
This is still an issue in 9.0.1

**Comment by mendrew** [comment]
I've tried to do research and found that it's true if component has **padding** applied.
I've been able to reproduce this directly on component with div around + padding on wrapper div, wrapper component didn't have styles.

I'm not sure, but I've seen an issue in react which is not the same but might be relevant https://github.com/facebook/react/issues/10906 and it says that some onMouseEnter incorrect calculation has been fixed in `16.1.0-beta`.

Why I'm talking about it if styleguidist uses newer version of react?

To compile markdown to jsx styleguidist uses [markdown-to-jsx](https://github.com/probablyup/markdown-to-jsx) module and it uses version `15.6.1` of react  https://github.com/probablyup/markdown-to-jsx/blob/9570018bac22c759899be839089cc33d3d739af0/package.json#L60 

Could it be possible that during compilation we are getting this incorrect calculation bug for onMouseEnter event because of complier react version?
Is compiler use styleguidist react version during execution? don't know.
I'm still far from understending of how it's all working together and affecting each other.
